typescript/sdk/src/providers/SmartProvider/SmartProvider.foundry-test.ts

```diff
@@ -142,4 +144,57 @@ describe('SmartProvider', async () => {
 
     expect(erc20.address).to.not.be.empty;
   });
+
+  it('returns the blockchain error reason: "ERC20: transfer to zero address"', async () => {
+    const smartProvider = HyperlaneSmartProvider.fromRpcUrl(NETWORK, URL, {
+      maxRetries: 1,
+    });
+    const signer = new Wallet(PK, smartProvider);
+
+    const factory = new ERC20__factory(signer);
+    const token = await factory.deploy('fake', 'FAKE');
+    try {
+      await token.transfer(constants.AddressZero, 1000000);
+    } catch (e: any) {
+      expect(e.error.message).to.equal(
+        'execution reverted: revert: ERC20: transfer to the zero address',
```

Comment:
nit: seems slightly redundant to have `reverted: revert: XXX`

Reply:
i believe that comes from ethers. i'm only passing that message